Ambiguous  call to `nvtx3::payload::payload()` on macOS

```c++
/Users/jfaibussowit/soft/nv/legate.core.internal/arch-io/cmake_build/_deps/kvikio-src/cpp/include/kvikio/posix_io.hpp:276:3: error: call to constructor of 'nvtx3::payload' is ambiguous
  276 |   KVIKIO_NVTX_FUNC_RANGE("posix_host_read()", size);
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/jfaibussowit/soft/nv/legate.core.internal/arch-io/cmake_build/_deps/kvikio-src/cpp/include/kvikio/utils.hpp:309:18: note: expanded from macro 'KVIKIO_NVTX_FUNC_RANGE'
  308 |   GET_KVIKIO_NVTX_FUNC_RANGE_MACRO(                                  \
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  309 |     __VA_ARGS__, KVIKIO_NVTX_FUNC_RANGE_2, KVIKIO_NVTX_FUNC_RANGE_1) \
      |     ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  310 |   (__VA_ARGS__)
      |   ~~~~~~~~~~~~~
/Users/jfaibussowit/soft/nv/legate.core.internal/arch-io/cmake_build/_deps/kvikio-src/cpp/include/kvikio/utils.hpp:286:61: note: expanded from macro 'GET_KVIKIO_NVTX_FUNC_RANGE_MACRO'
  286 | #define GET_KVIKIO_NVTX_FUNC_RANGE_MACRO(_1, _2, NAME, ...) NAME
      |                                                             ^
/Users/jfaibussowit/soft/nv/legate.core.internal/arch-io/cmake_build/_deps/kvikio-src/cpp/include/kvikio/utils.hpp:283:12: note: expanded from macro 'KVIKIO_NVTX_FUNC_RANGE_2'
  283 |       msg, nvtx3::payload { val }                             \
      |            ^              ~~~~~~~
```
On macOS, `std::size_t` is `unsigned long`, which -- crucially -- is not `std::uint64_t` (`unsigned long long`).

Perhaps consider doing
```c++
KVIKIO_NVTX_FUNC_RANGE("posix_host_read()", std::uint64_t{size});
```
whenever `size` is `std::size_t`. Alternatively, the macro could be amended to do this automatically.

--- a/cpp/include/kvikio/utils.hpp
+++ b/cpp/include/kvikio/utils.hpp
@@ -77,8 +77,10 @@ inline constexpr std::size_t page_size = 4096;
 template <typename T, std::enable_if_t<std::is_integral_v<T>>* = nullptr>
 [[nodiscard]] std::int64_t convert_to_64bit(T value)
 {
-  if (value >= std::numeric_limits<std::int64_t>::max()) {
-    throw std::overflow_error("convert_to_64bit(x): x too large to fit std::int64_t");
+  if constexpr (std::numeric_limits<T>::max() > std::numeric_limits<std::int64_t>::max()) {
+    if (value > std::numeric_limits<std::int64_t>::max()) {
+      throw std::overflow_error("convert_to_64bit(x): x too large to fit std::int64_t");
+    }
   }
   return std::int64_t(value);
 }

--- a/cpp/include/kvikio/utils.hpp
+++ b/cpp/include/kvikio/utils.hpp
@@ -21,7 +21,9 @@
 #include <iostream>
 #include <map>
 #include <optional>
+#include <stdexcept>
 #include <tuple>
+#include <type_traits>
 
 #include <nvtx3/nvtx3.hpp>
 
@@ -67,6 +69,27 @@ inline constexpr std::size_t page_size = 4096;
 {
   // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
   return reinterpret_cast<CUdeviceptr>(devPtr);
+}
+
+/**
+ * @brief Help function to convert value to 64 bit signed integer
+ */
+template <typename T, std::enable_if_t<std::is_integral_v<T>>* = nullptr>
+[[nodiscard]] std::int64_t convert_to_64bit(T value)
+{
+  if (value >= std::numeric_limits<std::int64_t>::max()) {
+    throw std::overflow_error("convert_to_64bit(x): x too large to fit std::int64_t");
+  }
+  return std::int64_t(value);
+}
+
+/**
+ * @brief Help function to convert value to 64 bit float
+ */
+template <typename T, std::enable_if_t<std::is_floating_point_v<T>>* = nullptr>
+[[nodiscard]] double convert_to_64bit(T value)
+{
+  return double(value);
 }
 
 /**
@@ -280,7 +303,7 @@ struct libkvikio_domain {
   {                                                           \
     nvtx3::event_attributes                                   \
     {                                                         \
-      msg, nvtx3::payload { val }                             \
+      msg, nvtx3::payload { convert_to_64bit(val) }           \
     }                                                         \
   }
 #define GET_KVIKIO_NVTX_FUNC_RANGE_MACRO(_1, _2, NAME, ...) NAME

